tests/UnitTests/ApplicationCore/Extensions/FooBar.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Microsoft.eShopWeb.UnitTests.ApplicationCore.Extensions
+{
+    public class Foo : IEquatable<Foo>
```

Comment:
Could we:

-  Give these some more meaningful names? Something like `TestParent` and `TestChild`? Something to be more descriptive of what the actual classes are.
- Split the two classes out in to their own files?